node/hyperbahn/service_proxy.js

```diff
@@ -295,11 +299,60 @@ ServiceDispatchHandler.prototype.refreshServicePeer =
 function refreshServicePeer(serviceName, hostPort) {
     var self = this;
 
+    // Add hostPort to service peers, decision pending on whether to connect.
     var peer = self.getServicePeer(serviceName, hostPort);
-    peer.connect();
 
+    // The old way: fully connect every egress to all affine peers.
+    if (!self.partialAffinityEnabled) {
+        peer.connect();
+        return;
+    }
+
+    // Make a note not to collect this service
     var time = self.channel.timers.now();
```

Comment:
we used to always do this, now we don't in the "off" case, please restore behavior?